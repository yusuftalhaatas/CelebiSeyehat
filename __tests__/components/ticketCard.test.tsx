import React from 'react';
import {render} from '@testing-library/react-native';

import TicketCard from '../../src/components/ticketCard';

describe('components/TicketCard', () => {
  test('renders correctly', () => {
    const ticketData = {
      date: '2023-05-15',
      start: 'Test Start',
      finish: 'Test Finish',
      firmName: 'Test Firm',
      no: 'A101',
      name: 'Test Name',
    };

    const {getByText, toJSON} = render(<TicketCard {...ticketData} />);

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // Check if all fields are displayed correctly
    expect(getByText(`Name: ${ticketData.name}`)).toBeDefined();
    expect(getByText(`Firm: ${ticketData.firmName}`)).toBeDefined();
    expect(getByText(`${ticketData.start}-${ticketData.finish}`)).toBeDefined();
    expect(getByText(`Date:${ticketData.date}`)).toBeDefined();
    expect(getByText(`Seat No:${ticketData.no}`)).toBeDefined();
  });
});
