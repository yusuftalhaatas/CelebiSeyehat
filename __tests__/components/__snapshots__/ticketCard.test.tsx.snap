// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TicketCard renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#00ccff",
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Name: 
    Test Name
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Firm: 
    Test Firm
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Test Start
    -
    Test Finish
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Date:
    2023-05-15
  </Text>
  <Text
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    Seat No:
    A101
  </Text>
</View>
`;
